gcloud-java-compute/src/test/java/com/google/gcloud/compute/AddressTest.java

```diff
@@ -0,0 +1,284 @@
+    assertSame(serviceMockReturnsOptions, address.compute());
+    assertNull(address.address());
+    assertNull(address.creationTimestamp());
+    assertNull(address.description());
+    assertNull(address.id());
+    assertNull(address.status());
+    assertNull(address.usage());
+    address = new Address.Builder(serviceMockReturnsOptions, REGION_ADDRESS_ID).build();
+    assertEquals(REGION_ADDRESS_ID, address.addressId());
+    assertSame(serviceMockReturnsOptions, address.compute());
+    assertNull(address.address());
+    assertNull(address.creationTimestamp());
+    assertNull(address.description());
+    assertNull(address.id());
+    assertNull(address.status());
+    assertNull(address.usage());
+  }
+
+  @Test
+  public void testToBuilder() {
+    initializeExpectedAddress(16);
+    compareAddress(instanceAddress, instanceAddress.toBuilder().build());
+    compareAddress(globalForwardingAddress, globalForwardingAddress.toBuilder().build());
+    compareAddress(regionForwardingAddress, regionForwardingAddress.toBuilder().build());
+    Address newAddress = instanceAddress.toBuilder().description("newDescription").build();
+    assertEquals("newDescription", newAddress.description());
+    newAddress = newAddress.toBuilder().description("description").build();
+    compareAddress(instanceAddress, newAddress);
+  }
+
+  @Test
+  public void testToAndFromPb() {
+    initializeExpectedAddress(20);
+    compareAddress(globalForwardingAddress,
+        Address.fromPb(serviceMockReturnsOptions, globalForwardingAddress.toPb()));
+    compareAddress(regionForwardingAddress,
+        Address.fromPb(serviceMockReturnsOptions, regionForwardingAddress.toPb()));
+    compareAddress(instanceAddress,
+        Address.fromPb(serviceMockReturnsOptions, instanceAddress.toPb()));
+    Address address = new Address.Builder(serviceMockReturnsOptions, GLOBAL_ADDRESS_ID).build();
+    compareAddress(address, Address.fromPb(serviceMockReturnsOptions, address.toPb()));
+    address = new Address.Builder(serviceMockReturnsOptions, REGION_ADDRESS_ID).build();
+    compareAddress(address, Address.fromPb(serviceMockReturnsOptions, address.toPb()));
+  }
+
+  @Test
+  public void testDeleteOperation() {
+    initializeExpectedAddress(4);
+    expect(compute.options()).andReturn(mockOptions);
+    Operation operation = new Operation.Builder(serviceMockReturnsOptions)
+        .operationId(GlobalOperationId.of("project", "op"))
+        .build();
+    expect(compute.delete(REGION_ADDRESS_ID)).andReturn(operation);
+    replay(compute);
+    initializeAddress();
+    assertSame(operation, address.delete());
+  }
+
+  @Test
+  public void testDeleteNull() {
+    initializeExpectedAddress(3);
+    expect(compute.options()).andReturn(mockOptions);
+    expect(compute.delete(REGION_ADDRESS_ID)).andReturn(null);
+    replay(compute);
+    initializeAddress();
+    assertNull(address.delete());
+  }
+
+  @Test
+  public void testExists_True() throws Exception {
```

Comment:
I do not really mind it, I in fact prefer this when the name gets super long. However, is this a common practice when naming tests? I have never seen it before.